infrastructure/cognito/cognito.tf

```diff
@@ -0,0 +1,48 @@
+# ----------------------------------------------------------------------------------------------------------------------
+# CREATE USER POOL 
+# @param alias_attributes Attributes supported as an alias for the user pool. Valid values: phone_number, email, or preferred_username
+# @param generate_secret Should an application secret be generated.
+# ----------------------------------------------------------------------------------------------------------------------
+
+# ----------------------------------------------------------------------------------------------------------------------
+# USER POOL
+# ----------------------------------------------------------------------------------------------------------------------
+resource "aws_cognito_user_pool" "pool" {
+  name = "${var.environment}_pool"
+  alias_attributes = ["email"]
+}
+
+# ----------------------------------------------------------------------------------------------------------------------
+# USER GROUPS
+# ----------------------------------------------------------------------------------------------------------------------
+
+resource "aws_cognito_user_group" "admin" {
+  name         = "${var.environment}_admin_group"
+  user_pool_id = aws_cognito_user_pool.pool.id
+  description  = "Administrator group"
+}
+
+resource "aws_cognito_user_group" "customer" {
+  name         = "${var.environment}_customer_group"
+  user_pool_id = aws_cognito_user_pool.pool.id
+  description  = "Customer group"
+}
+
+# ----------------------------------------------------------------------------------------------------------------------
+# CLIENTS
+# ----------------------------------------------------------------------------------------------------------------------
+
+resource "aws_cognito_user_pool_client" "client" {
```

Comment:
Can you research which will be the default value for each unit used?
Or use the [refresh_token_validity, id_token_validity and access_token_validity](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cognito_user_pool_client#access_token_validity) instead.